### DISCUSSION TITLE
类似搜狗拼音的拆字反查，是否可通过导入已有 prism 来支持双拼

### DISCUSSION THREAD

**Discussion by functoreality** [rime/home#1091] [Q&A]
搜狗输入法里用 u 前缀可进入拆字模式。例如，在全拼下敲 `usichou` 可以输入“纽”，在小鹤双拼下则是敲 `usiiz`，其他双拼方案下输入码也有相应变化。
现在希望在 Rime 里实现类似的功能。拆字字典 `chaizi.dict.yaml` 与[这里](https://github.com/rime/home/discussions/764)所用的类似，格式如下（编译字典所用的相应 schema 文件略去）：
```yaml
纽	si chou	1
```
在 `luna_pinyin.custom.yaml` 添加如下的行，则可以在反查模式下敲 `sichou` 输入“纽”：
```yaml
reverse_lookup/dictionary: chaizi
```
但是在 `double_pinyin_flypy.custom.yaml` 中添加如下的行：
```yaml
reverse_lookup/dictionary: chaizi
reverse_lookup/prism: double_pinyin_flypy
```
期待的行为是在反查模式下敲 `siiz` 能输入“纽”，而实际上什么输出也没有。
目前推测问题在于生成的 prism 文件除了拼写运算以外还包括了码表信息，无法直接用于其他的码表。如果专门再写一个新的 schema 文件生成 `chaizi_flypy.prism.bin` 替换掉自带小鹤双拼的棱镜文件，则确实能按照预期的模式工作。但是为不同双拼方案分别编写相应的 schema 文件稍显低效，何况所用到的拼写运算都是已有的。想知道是否可能直接用现成的棱镜文件达到目的，或者有其他更加简捷的实现方式。

**Comment by lotem** [comment]
> 但是为不同双拼方案分别编写相应的 schema 文件稍显低效，何况所用到的拼写运算都是已有的。

錯. 棱鏡文件要與匹配的固態詞典配套使用.
所謂"低效", "用到的拼寫運算都是已有的", "現成的棱鏡文件", 只是不知情人的錯覺, 或者說人文領域的知識.
對邏輯嚴密的人或程序來說, 拼音詞典和反查詞典只是恰好有部分編碼相同的兩部詞典, 不能混爲一談. 舉例來說, 拼音詞典中可能存在更多的拼音音節, 於是一條拼寫映射規則可能在拼音詞典中生成的對應關係是 `e = [e, ei, en, eng]` 而反查詞典裏可能對應爲 `e = [e, en]`, 因爲一些音節在該詞典中不存在. 如果棱鏡文件中包含到詞典中不存在的編碼的映射, 怎麼能夠認爲能與該詞典匹配呢.

> 想知道是否可能直接用现成的棱镜文件达到目的，或者有其他更加简捷的实现方式。

如上所述, 根本沒有現成的棱鏡文件. 兩者採用相同的編碼規則屬於作者的意識範疇, 對程序而言不會因爲二者包含一些相同的編碼而把他們視爲同一種輸入法.
不可以復用針對具體詞典編譯的二進制文件. 僅可以在源碼級別復用, 即共同引用實現雙拼拼寫運算的YAML代碼片段.

**Reply by functoreality** [reply]
> 舉例來說, 拼音詞典中可能存在更多的拼音音節, 於是一條拼寫映射規則可能在拼音詞典中生成的對應關係是 `e = [e, ei, en, eng]` 而反查詞典裏可能對應爲 `e = [e, en]`, 因爲一些音節在該詞典中不存在. 

感谢解读，看来我的猜测基本是成立的。之前看的文档都只解释了抽象出来的接口的用法，对其底层的逻辑实现方式一直不甚了解，而体量庞大的 librime 代码也不知道从何读起。
